src/java.base/share/classes/com/sun/crypto/provider/RSACipher.java

```diff
@@ -349,21 +349,38 @@ private byte[] doFinal() throws BadPaddingException,
             switch (mode) {
```

Comment:
Nothing wrong what you did here, but since you used the new style switch elsewhere, you could do:
```
return switch(mode) {
    case MODE_SIGN -> {
            paddingCopy = padding.pad(buffer, 0, bufOfs);
            if (paddingCopy == null) {
                throw new BadPaddingException("Padding error in signing");
            }
            yield RSACore.rsa(paddingCopy, privateKey, true);
     }
```
... and so on...

Reply:
In code that would probably need to be back ported, I would not use language features that might not be available in earlier versions (I don't know whether this is the case here, though) unless they provide noticeable performance benefits, not just "look better/more modern".